woocommerce/woocommerce-ios
# Product Settings: edit the Purchase Note

Resolve #2121 

This PR implement the edit product purchase note under Product Settings, part of milestone 2.

## Changes
- Encoded the `purchaseNote` field in Product model
- New Product Setting row `PurchaseNote`
- Implemented the new `ProductPurchaseNoteViewController`
- Implemented the new `EnhancedTextView` that support the placeholder label inside the text view, and implement the new `onTextChange:` closure.
- Refactored `TextViewTableViewCell` that now uses the `EnhancedTextView` and that allows us to hide the note icon.
- Updated tests to support the new `purchaseNote` field
- Updated the `ProductUpdated`

## Testing
1. Navigate to a product detail
2. Press the "more" button in the navigation bar
3. Tap "Product Settings" -> You are able to see the cell "Purchase Note" with the current purchase note.
4. Tap the "Purchase Note" cell, you will be able to see the current purchase note. A placeholder in case there is no content.
5. Try to change the purchase note. Press back.
6. Press "done", and update the product. -> Reopening the Purchase Note view, you will be able to see the new purchase note updated.

## Screenshots
| Product Settings             |  Empty Purchase Note |  populated Purchase Note  |
:-------------------------:|:-------------------------:|:-------------------------:
![Simulator Screen Shot - iPhone Xs - 2020-04-15 at 13 16 50](https://user-images.githubusercontent.com/495617/79332321-ed950680-7f1c-11ea-92dc-64c06d734122.png) | ![Simulator Screen Shot - iPhone Xs - 2020-04-15 at 13 16 28](https://user-images.githubusercontent.com/495617/79332381-07364e00-7f1d-11ea-850b-cadbe89d57cc.png) | ![Simulator Screen Shot - iPhone Xs - 2020-04-15 at 13 16 44](https://user-images.githubusercontent.com/495617/79332384-09001180-7f1d-11ea-97a4-ac5baa562316.png)



Update release notes:
